db/db_impl/db_impl_write.cc

```diff
@@ -924,6 +924,38 @@ Status DBImpl::WriteImplWALOnly(
       write_thread->ExitAsBatchGroupLeader(write_group, status);
       return status;
     }
+  } else {
+    // TODO(yanqin): maybe move this block into a refactored version of
+    // DelayWrite() so that we account for manually-injected write stalls and
+    // other write stalls separately.
```

Comment:
Can we call `DelayWrite()` and let it count manually-injected write stalls towards the "rocksdb.stall.micros" counter? It seems that's already the case for the other case (`publish_last_seq == kDoPublishLastSeq`) where it calls `PreprocessWrite()`, which calls `DelayWrite()`.

Reply:
Thought about this. The reason I didn't re-use `DelayWrite()` is because I am not sure if it's desirable for the specific customer to rate limit WAL writes. Therefore, I only checked for stop token.

Reply:
Oh.. That is too bad there are different classes of write stalls. Is there any way to unify without double-throttling write-prepared (on both prepare here and later commit)?

Reply:
> Oh.. That is too bad there are different classes of write stalls. Is there any way to unify without double-throttling write-prepared (on both prepare here and later commit)?

@riversand963 How about call `DelayWrite(0 /* num_bytes */, ...)` for this case?

Reply:
Good point. I misread the code and somehow thought passing 0 as `nums_bytes` would cause the function not to check for write stop. Updated.